docs/source/api/algorithm_functions/index.rst

```diff
@@ -0,0 +1,19 @@
+.. _algorithm_api:
+
+Algorithm Functions
+===================
+
+.. toctree::
+   :maxdepth: 1
```

Comment:
I think making this depth 2 would be nice so we still retain the list of functions. Especially in the move to furo depth 2 would be needed to show each function in the side bar right? But also for the current pytorch based theme I think having the full function list on the api page is useful, even if a bit overwhelming at times.

Reply:
`maxdepth` has zero impact on Furo's left sidebar. This is how it renders even with maxdepth set to 1

![Screenshot 2023-07-12 at 10 15 48 AM](https://github.com/Qiskit/rustworkx/assets/14852634/ddbb1dfb-234a-4754-bcb1-6db25486dc60)

> I think making this depth 2 would be nice so we still retain the list of functions...I think having the full function list on the api page is useful, even if a bit overwhelming at times.

Sounds reasonable to me. We can always change it with one line if we don't like it.